src/dialog-user/components/dialog-user/dialogField.html

```diff
@@ -61,12 +61,8 @@
               ng-disabled="vm.dialogField.read_only || vm.inputDisabled"
               class="form-control"
               data-container="body"
-              id="{{ vm.dialogField.name }}">
-        <option ng-repeat="value in vm.dialogField.values track by $index"
-                data-tokens="{{value[0]}} {{value[1]}}"
-                value="{{value[0]}}">
-          {{value[1]}}
-        </option>
+              id="{{ vm.dialogField.name }}"
+              ng-options="value[0] as value[1] for value in vm.dialogField.values">
```

Comment:
Well, this is a bit of a problem because we need `data-tokens`. We had the `ng-options` before, but it does not allow the in-dropdown search function to work without the `data-tokens`, and there's no way that I know of to establish them without doing the `ng-repeat`.

Reply:
What do you need it for? There is no code in ui-classic, ui-components, or ui-classic dependencies which would look at data-tokens.

Reply:
(Could it be that this was only needed back when angular-patternfly 4 pf-select was used?)

Reply:
But if it is still needed, if we can be sure the server doesn't care about types, we could convert everything to string and it could work too.

It's just that I don't see any user of this now.

Reply:
It's for the dropdown search functionality (`data-live-search="true"`), if a customer has a big huge long list of things in their dropdown, they can type to do a search for things. The problem is that without `data-tokens`, I believe it either searches on the value or the description (can't remember which one), but it needs to be able to search on both, which is why the `data-tokens` attribute needs to exist.

Reply:
OK, but, `@pf3/select` does not touch `data-tokens`, and neither does the `miq-select` directive.

So.. I guess what you're saying is that we need to backport the `data-tokens` logic to `@pf3/select`.

Reply:
To be clear about this, I don't think `data-tokens` ever worked in ui-classic, but they may have worked in service ui.

Reply:
@himdel pf is based off of bootstrap as far as I know, and the live-search functionality has been available in dropdowns for a while now. Here's some documentation [from the patternfly side](https://github.com/patternfly/patternfly.github.io/blob/master/components/bootstrap-select/docs/docs/examples.md#live-search) and [from the bootstrap side](https://developer.snapappointments.com/bootstrap-select/examples/#live-search)

Reply:
Yeah, sorry about that, I was searching wrong. I can see the code now...

https://github.com/himdel/bootstrap-select/blob/master/js/bootstrap-select.js#L185-L206

So, yeah, this worked, and I'm looking for a way of forcing ng-options to create that..

Reply:
... found one :)

![teehee](https://user-images.githubusercontent.com/289743/55436363-9d563b00-558b-11e9-8066-24c575f4426c.png)

Reply:
Good find 👍